### ISSUE TITLE
Regexp allow invalid Orcid prefix 

### ISSUE THREAD

**Issue by leomendoza123** [ORCID/orcid-angular#169]
Regexp ORCID_REGEXP and ORCID_URI_REGEXP to validate ORCID IDs allow any string at the begging

https://github.com/ORCID/orcid-angular/blob/master/src/app/constants.ts

This allows inputs to accept wrong inputs like: 

`https://INVALID.orcid.org/0000-0002-9361-1905` or `INVALID!0000-0002-9361-1905`

This will currently affect validation on the advance search and signing 

![image](https://user-images.githubusercontent.com/2119626/80624875-8a6d9e80-8a09-11ea-9764-0f5191f26f29.png)

**Comment by DanielPalafox** [comment]
Currently, in the actual UI in production, you can put anything before the ORCID ID as Leo explains. We just need to accept valid ORCID ids or valid full URLs and with this, we will avoid making unnecessary calls to the backend.

**Comment by leomendoza123** [comment]
Yes on the old sign in it will even allow you to successfully sign with an invalid string (if it contains a valid Orcid Id)

![image](https://user-images.githubusercontent.com/2119626/80628390-82642d80-8a0e-11ea-888d-55080bd4c46b.png)

We should decide if we really want to validate a perfect ORCID ID for the signin since the backend understands the call, so maybe we want to keep allowing that there. 

On the other hand, I believe we need a good validation on most other features.
